zaza/model.py

```diff
@@ -125,31 +110,25 @@ def scp_to_all_units(application_name, model_name, source, destination,
     :param scp_opts: Additional options to the scp command
     :type scp_opts: str
     """
-    async def _scp_to_all_units(application_name, source, destination, user,
-                                proxy, scp_opts, model):
-        for unit in model.applications[application_name].units:
-            await unit.scp_to(source, destination, user=user, proxy=proxy,
-                              scp_opts=scp_opts)
-    scp_func = functools.partial(
-        _scp_to_all_units,
-        application_name,
-        source,
-        destination,
-        user=user,
-        proxy=proxy,
-        scp_opts=scp_opts)
-    loop.run(
-        run_in_model(model_name, scp_func, add_model_arg=True, awaitable=True))
-
-
-def scp_from_unit(unit_name, model_name, source, destination, user='ubuntu',
+    async def _scp_to_all_units(model_name, application_name, source,
+                                destination, user, proxy, scp_opts):
+        async with RunInModel(model_name) as model:
+            for unit in model.applications[application_name].units:
```

Comment:
Will this end up `await`ing each unit individually, or queuing them all up and `await`ing them as a batch?